sdk/keyvault/azure-security-keyvault-certificates/test/perf/inc/azure/keyvault/certificates/test/get_certificate_test.hpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Test the overhead of getting a certificate.
+ *
+ */
+
+#pragma once
+
+#include <azure/perf.hpp>
+
+#include <azure/core/internal/environment.hpp>
+#include <azure/identity.hpp>
+#include <azure/keyvault/certificates.hpp>
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+
+using namespace Azure::Core::_internal;
+
+namespace Azure {
+  namespace Security {
+    namespace KeyVault {
+      namespace Certificates {
+        namespace Test {
+
+  /**
+   * @brief A test to measure getting a key performance.
+   *
+   */
+  class GetCertificate : public Azure::Perf::PerfTest {
+  private:
+    std::string m_vaultUrl;
+    std::string m_certificateName;
+    std::string m_tenantId;
+    std::string m_clientId;
+    std::string m_secret;
+    std::shared_ptr<Azure::Identity::ClientSecretCredential> m_credential;
+    std::unique_ptr<Azure::Security::KeyVault::Certificates::CertificateClient> m_client;
+
+  public:
+    /**
+     * @brief Get the Ids and secret
+     *
+     */
+    void Setup() override
+    {
+      m_vaultUrl = m_options.GetOptionOrDefault<std::string>(
+          "vaultUrl", Environment::GetVariable("AZURE_KEYVAULT_URL"));
+      m_tenantId = m_options.GetOptionOrDefault<std::string>(
+          "TenantId", Environment::GetVariable("AZURE_TENANT_ID"));
+      m_clientId = m_options.GetOptionOrDefault<std::string>(
+          "ClientId", Environment::GetVariable("AZURE_CLIENT_ID"));
+      m_secret = m_options.GetOptionOrDefault<std::string>(
+          "Secret", Environment::GetVariable("AZURE_CLIENT_SECRET"));
```

Comment:
That's interesting.
User can pass command line args directly for running the test, or they can set the env.

What happen if they don't do neither?  I think the client will fail to connect right?  

SHould we validate that values are not empty or throw and tell users they forgot to either set the env or user command args?

Reply:
i was thinking about it, but it will fail creating the client anyway m so we follow the we don't validate params idea from the sdk